src/test/service/GeocodingService.test.ts

```diff
@@ -0,0 +1,57 @@
+import { GeocodingService } from '../../main/service/GeocodingService';
+import { GeocodingObject } from '../../main/data/GeocodingObject';
+import { randomString } from '../utility/Utility';
+
+describe('tests for Geocoding Service', function () {
+  let geocoder: GeocodingService;
+
+  beforeAll(() => {
+    geocoder = new GeocodingService();
```

Comment:
I think it's not that good to use API keys for tests,  That's why I used a mock for the Google place service test. 

I think there was a free provider that didn't require an api key, thing it was called open maps or something. We could make geocodingservice constructor take in the configuration  for node geocoder, or make it take in a node geocoder instance too

Another thing we could do is see if we can find a mocking library for node geocoder. Last resort would be to make the mock class ourself!

Reply:
I considered the mock you had but I think it's fine considering that we're in the free tier, there's no storage of info and no chance we'll get charged for over usage

Reply:
Hmm let's picture it this way:

the free tier has 60 requests per minute
Currently, in our test class, we make 5 requests
Let's say theoretically we have 6 PRs currently up and running integration tests at the same time - 30 requests left
At the same time 2 developers are developing locally and each test the geocoding 10 times - 10 requests left
And 100 customers are uploading promotions (sadly no more requests left until we wait another minute)

Similar reasons can be seen for the 5000 request limit
- https://softwareengineering.stackexchange.com/questions/184979/unit-testing-open-source-projects-that-require-keys-for-api-calls
- https://softwareengineering.stackexchange.com/questions/214154/how-can-i-unit-test-a-class-which-requires-a-web-service-call/214159#:~:text=Your%20unit%20test%20should%20not,you%20want%20fast%20unit%20tests.

Reply:
Ahh this is a good point. If we can mock it out, I think that would be better practice